src/main/scala/swiss/dasch/domain/IngestService.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright © 2021 - 2023 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package swiss.dasch.domain
+
+import eu.timepit.refined.types.string.NonEmptyString
+import org.apache.commons.io.FilenameUtils
+import swiss.dasch.domain.Asset.{OtherAsset, StillImageAsset}
+import swiss.dasch.domain.DerivativeFile.OtherDerivativeFile
+import swiss.dasch.domain.PathOps.fileExtension
+import zio.nio.file.Path
+import zio.{IO, Task, ZIO, ZLayer}
+
+import java.io.IOException
+
+final case class IngestService(
+  storage: StorageService,
+  imageService: ImageService,
+  sipiClient: SipiClient,
+  assetInfo: AssetInfoService
+) {
+
+  def ingestFile(fileToIngest: Path, project: ProjectShortcode): Task[Asset] =
+    for {
+      _ <- ZIO.logInfo(s"Ingesting file $fileToIngest")
+      _ <- ZIO.whenZIO(FileFilters.isSupported(fileToIngest).negate)(
```

Comment:
Is there a particular reason you prefer
1. `ZIO.whenZIO(condition)(doSomething)` over `doSomething.whenZIO(condition)`?
2. `ZIO.whenZIO(condition.negate)(doSomething)` over `ZIO.unlessZIO(condition)(doSomething)`
if it's merely stylistic, then I don't care much; if you have a functional reason, I'd like to learn it

Reply:
OK. I didn't mean to impose any stylistics, but if you like it that way then I'm glad because I find the `.negate` easy to miss